app/src/composables/use-preset.ts

```diff
@@ -88,6 +88,14 @@ export function usePreset(
 
 	function updatePreset(preset: Partial<Preset>, immediate?: boolean) {
 		localPreset.value = assign({}, localPreset.value, preset);
+
+		const hasChanged = Object.keys(preset.layout_query ?? {}).some((key) => {
+			const query = (preset.layout_query ?? {})[key];
+			return !(Object.keys(query).length === 1 && 'page' in query);
+		});
+
+		if (!hasChanged) return;
+
 		immediate ? savePreset() : handleChanges();
 	}
```

Comment:
```suggestion
	function checkIfChanged(existing: unknown, updated: unknown) {
		if (typeof existing !== 'object' || existing === null || typeof updated !== 'object' || updated === null) {
			return existing !== updated;
		}

		for (const key in updated) {
			if (key in existing === false) return true;
			const changed = checkIfChanged(existing[key], updated[key]);

			if (changed) {
				return true;
			}
		}

		return false;
	}

	function updatePreset(preset: Partial<Preset>, immediate?: boolean) {
		const hasChanged = checkIfChanged(localPreset.value, preset);
		if (!hasChanged) return;

		localPreset.value = assign({}, localPreset.value, preset);
		immediate ? savePreset() : handleChanges();
	}
```

Reply:
Won't this still save presets when the page changes? The new preset coming into `updatePreset` has a single key called `page` which i was trying to ignore

Reply:
@br41nslug Here is a recording of the fix i implemented, it appears to be working or is there a specific case that doesn't work?

[screen-capture.webm](https://github.com/directus/directus/assets/15183853/a4daa908-05de-4253-acf0-3fa7c8559597)

Reply:
Not sure if this is the right place to handle this. If possible I'd solve this at the `layout` level, so it doesn't get passed to `updatePreset()` / isn't part of layout query in the first place. 

That being said, taking a step back, I think we need to decide whether we indeed want drop `page` from the presets. While that probably makes sense, I still feel like it's worth a separate issue, especially as it may have been used by some users so far. This is also reflected in the fact that we are already having a major discussion here.

The other two changes included in this PR (render order in calendar & check for change of `selectedSize`) are clearly bug fixes and could already be merged by now, IMO.

Reply:
> Not sure if this is the right place to handle this. If possible I'd solve this at the `layout` level, so it doesn't get passed to `updatePreset()` / isn't part of layout query in the first place.

That would make sense, as the page should be separate from layout in my opinion. 

> That being said, taking a step back, I think we need to decide whether we indeed want drop `page` from the presets. While that probably makes sense, I still feel like it's worth a separate issue, especially as it may have been used by some users so far. This is also reflected in the fact that we are already having a major discussion here.

We could open a discussion and ask more people, I reckon the biggest argument for that is the dynamic nature of pages and whether saving the current page actually makes sense. 

> The other two changes included in this PR (render order in calendar & check for change of `selectedSize`) are clearly bug fixes and could already be merged by now, IMO.

I agree

Reply:
> That would make sense, as the page should be separate from layout in my opinion.

This definitely needs a bit more thinking. The original intent for persisting that information is persisting your place in the navigation when navigating around / logging in elsewhere. The current page of results is pretty important in that still, as you wouldn't want to be back on page 1 when you navigate in and out of a record 🤔

Reply:
> > That would make sense, as the page should be separate from layout in my opinion.
> 
> This definitely needs a bit more thinking. The original intent for persisting that information is persisting your place in the navigation when navigating around / logging in elsewhere. The current page of results is pretty important in that still, as you wouldn't want to be back on page 1 when you navigate in and out of a record 🤔

I wonder if we could just save an actual "local" preset in local storage? That would keep persistence in a session without saving a new preset which is very annoying in a multi user app

Reply:
Worth thinking through! Lets discuss that in a feature request discussion though instead of a feature request that already partially implements something we aren't really sure about yet 😄

Reply:
Yeah lets do it! The rest is looking good to me 🙂

Reply:
> so should I remove the change about the page? 

reverted via https://github.com/directus/directus/commit/bea5cf26122e9f9940588c0da21a1d64e08ee64d